crates/polars-lazy/src/scan/ipc.rs

```diff
@@ -96,11 +97,17 @@ impl LazyFileListReader for LazyIpcReader {
     }
 
     fn with_path(mut self, path: PathBuf) -> Self {
-        self.path = path;
+        self.path = resolve_homedir(&path);
         self
     }
 
     fn with_paths(mut self, paths: Arc<[PathBuf]>) -> Self {
+        let paths = paths
+            .iter()
+            .map(|p| resolve_homedir(p))
+            .collect::<Vec<_>>()
+            .into();
```

Comment:
Not entirely sure, but I believe we can collect into a `Arc<[]>` directly.

Reply:
Ahhh; think an `into_iter()` with a `&p` should optimise appropriately.
Updated 🤔

Reply:
Hmm, can't see a way to avoid the intermediate `Vec`; the `Arc` needs a constructed object to reference 🤔 If you can spot something cunning, feel free to point me at it or commit over the top :))

Reply:
Now I am sure. :D

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=82ca9fa7f2c267ad4811d7714747b7ae

Reply:
> Now I am sure. :D
> 
> https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=82ca9fa7f2c267ad4811d7714747b7ae

Lol... pretty sure that the `collect` there still creates an intermediate `Vec`, it just gets deallocated after conversion to `Arc[i32]`  :)

Reply:
Yes, but Rust might optimize that later.